### PR TITLE
Inventory & Trade

### PR DESCRIPTION
**Pull request by swpnit82**
## Summary

A summary of changes being made in this PR

## Checklist
- [ ] Pre-push checks pass `npm run check`
  - [ ] Linter passing via `npm run lint`
  - [ ] Unit & Integration tests passing via `npm run test:packages`
  - [ ] Docker build process passing via `npm run build-client`
- [ ] If this PR is still a WIP, convert to a draft 
- [ ] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewers

## References

References to pertaining issue(s)

## QA Steps

1. `git checkout pr_branch_name`
2. `npm install`
3. `npm run dev-reinit`
4. `npm run dev`

List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos.

## Reviewers

@HexaField @speigg @NateTheGreatt

### CODE REVIEW

**Comment by kinjalravel** [comment]
@swpnit82 @HexaField @shawwwwwwww  I have followed QA Steps, there are multiple issues

1. User profile its not loading including User name, User Id etc.
2. Clicking on My Inventory button nothing load and page just got blank
3. Clicking on My Trading button nothing load and page just got blank
4. Clicking on My Wallet button nothing load and page just got blank

I have attached QA video here, tack a look at it.

https://user-images.githubusercontent.com/41296632/144366472-02f1c1c2-73b8-4b4c-b97a-5dcb9a548415.mov

**Comment by HexaField** [comment]
I have fixed some issues. Please QA again @kinjalravel

File: packages/server-core/src/user/identity-provider/identity-provider.class.ts [hunk 1]
```diff
@@ -153,7 +153,23 @@ export class IdentityProvider extends Service {
       },
       params
     )
+    // DRC
+    try {
+      if (result.user.userRole === 'user') {
```
Comment (barankyle) [inline packages/server-core/src/user/identity-provider/identity-provider.class.ts:158]: I think this should also occur if result.user.userRole === 'admin'.

File: packages/client-core/src/user/components/UserMenu/menus/ProfileMenu.tsx [hunk 2]
```diff
@@ -303,6 +306,15 @@ const ProfileMenu = (props: Props): any => {
                 {t('user:usermenu.profile.inviteCode')}: {selfUser.inviteCode.value}
               </h2>
             )}
+            <button onClick={() => history.push(`/inventory/${selfUser.id.value}`)} className={styles.walletBtn}>
```
Comment (barankyle) [inline packages/client-core/src/user/components/UserMenu/menus/ProfileMenu.tsx:309]: Should these buttons only appear for non-guests?

File: packages/client-core/src/user/components/UserMenu/menus/TradingContent.tsx [hunk 3]
```diff
@@ -0,0 +1,804 @@
...
+    setState((prevState) => ({
+      ...prevState,
+      selectedtype: id
+    }))
+    handleClose()
+  }
+
+  useEffect(() => {
+    if (data.length !== 0) {
+      console.log('DAATA ', data)
```
Comment (barankyle) [inline packages/client-core/src/user/components/UserMenu/menus/TradingContent.tsx:171]: Remove

File: packages/client-core/src/user/components/UserMenu/menus/TradingContent.tsx [hunk 4]
```diff
@@ -0,0 +1,804 @@
...
+          setState((prevState: any) => ({
+            ...prevState,
+            // url: data[0].url,
+            // metadata: data[0].metadata,
+            selectedid: inventory[0].user_inventory.userInventoryId,
+            inventoryList: [...inventory]
+          }))
+        } else {
+          let filtereddata = inventory.filter((val) => val.inventoryItemTypeId === selectedtype)
+          console.log(filtereddata, selectedtype)
```
Comment (barankyle) [inline packages/client-core/src/user/components/UserMenu/menus/TradingContent.tsx:195]: Remove

File: packages/client-core/src/user/components/Inventory/inventory.tsx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+      ...prevState,
+      isLoadingtransfer: true
+    }))
+    try {
+      const response = await client.service('user-inventory').patch({
+        userId: ids,
+        userInventoryId: itemid
+      })
+      fetchInventoryList()
+      console.log('success')
```
Comment (barankyle) [inline packages/client-core/src/user/components/Inventory/inventory.tsx:49]: Remove this and all other debugging console.logs that were introduced.

File: packages/client-core/src/user/components/Wallet/wallet.tsx [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+  const { t } = useTranslation()
+  const [state, setState] = useState<any>({
+    coinData: [],
+    walletData: [],
+    data: [],
+    coinlimit: 0,
+    coinDataReceive: [],
+    dataReceive: [],
+    walletDataReceive: [],
+    user: [],
```
Comment (barankyle) [inline packages/client-core/src/user/components/Wallet/wallet.tsx:24]: This should be renamed 'users'.

**Review by barankyle — APPROVED** [review]

**Pull request merged**